workers/generic-worker/run_task_as_current_user.go

```diff
@@ -50,8 +50,19 @@ func (r *RunTaskAsCurrentUserTask) RequiredScopes() scopes.Required {
 }
 
 func (r *RunTaskAsCurrentUserTask) Start() *CommandExecutionError {
+	r.resetPlatformData()
+	r.setTaskUserCredentialsEnvVar()
+	r.removeXDGRuntimeDirEnvVar()
```

Comment:
maybe this should be `r.platformSpecificActions()`, and linux uses it to remove XDG_RUNTIME_DIR and windows just has nothing to do? to avoid having a method about XDG runtime in windows code...